### ISSUE TITLE
Flaky signin: Cannot read properties of undefined (reading 'getApps'

### ISSUE THREAD

**Issue by borisyordanov** [nearform/playwright-firebase#180]
A lot of the time the package is working, but occasionally it fails. It appears that `getApps()` is run before the firebase scripts are inserted in the dom and evaluted.

![image](https://github.com/nearform/playwright-firebase/assets/12392666/fd6ea485-0418-40f4-97ad-7451be70dfa0)

**Comment by vikasbhandari2** [comment]
Thanks for reporting!

Can you provide steps to reproduce? We often need a [reproducible example](https://stackoverflow.com/help/minimal-reproducible-example), e.g. some code that allows someone else to recreate your problem by just copying and pasting it. If it involves more than a couple of different file, create a new repository on GitHub and add a link to that.

**Comment by Ademsk1** [comment]
Agreed with @vikasbhandari2 . Would be interested to know if you're passing through a specific version of Firebase as well.

**Comment by borisyordanov** [comment]
Unfortunately, this one is very sporadic - it doesn't happen in a specific test. On some runs it doesn't happen at all. 
Thankfully playwright's auto-retry helps make this bug something we can live with.

We're using firebase@10.5.2, but upgrading to 10.7.2 didn't solve the issue.

Are you sure that when you try to [get the firebase app](https://github.com/nearform/playwright-firebase/blob/master/plugin/Authentication.ts#L56) the script is inserted and that the cdn has sent the code over? There might be a race condition.
